snuba_sdk/query_visitors.py

```diff
@@ -335,7 +350,11 @@ def _combine(self, query: main.Query, returns: Mapping[str, str]) -> str:
         if query.legacy:
             body["legacy"] = query.legacy.value
         if query.parent_api:
-            body["parent_api"] = query.parent_api.name
+            body["parent_api"] = query.parent_api.value
+        if query.team:
+            body["team"] = query.team.value
+        if query.feature:
+            body["feature"] = query.feature.value
```

Comment:
Something that bothers me about this:

Why are these flags expressions when they're not part of the query? They're attached to the query but then we move it outside the query when we serialize it. Like theoretically you can have a subquery with flags even though that won't be honored (since it doesn't make sense). Why is it done this way?

Reply:
It's essentially a consequence of the legacy query and how this SDK was integrated. Since this library was added as a way to replace just the "query" part of the request, I only added one top level object, the `Query`. However it became clear as I was working that this library would also need to be able to handle the extra flags as well, particularly for converting legacy queries into SnQL.

I agree it doesn't make sense and it needs to be refactored. As we've expanded the flags this model makes less and less sense and adds to our tech debt. I think particularly if we make the SDK responsible for building the request body separately from the Query, we can better handle these flags.